packages/app-builder/src/components/ui/Toolbar/ToolbarItem.tsx

```diff
@@ -0,0 +1,146 @@
+    opacity: 0.5,
+    backgroundColor: '#bfbfbf',
+  },
+  mark: {
+    backgroundColor: '#bfbfbf',
+    height: 8,
+    width: 1,
+    marginTop: -3,
+  },
+  markActive: {
+    opacity: 1,
+    backgroundColor: 'currentColor',
+  },
+})(Slider)
+
+export type ToolbarItemProps = {
+  prefix?: string
+  label?: string
+  full?: boolean
+  propKey: string
+  index?: number
+  children?: React.ReactNode
+  type: string
+  onChange?: (value: any) => any
+}
+export const ToolbarItem = ({ full = false, propKey, type, onChange, index = 0, ...props }: ToolbarItemProps) => {
+  const {
+    actions: { setProp },
+    propValue,
+  } = useNode((node) => ({
+    propValue: node.data.props[propKey],
+  }))
+  const isArray = Array.isArray(propValue)
+  const value = isArray ? propValue[index] : propValue
+
+  return (
+    <Grid item xs={full ? 12 : 6}>
+      <div className="mb-2">
+        {['text', 'color', 'bg', 'number'].includes(type) ? (
+          <ToolbarTextInput
+            {...props}
+            type={type}
+            value={value}
+            onChange={(value) => {
+              setProp((props: any) => {
+                if (type === 'number') {
+                  value = parseInt(value)
+                }
+                if (isArray) {
+                  props[propKey][index] = onChange ? onChange(value) : value
+                } else {
+                  props[propKey] = onChange ? onChange(value) : value
+                }
+              }, 500)
+            }}
+          />
+        ) : type === 'slider' ? (
+          <>
+            {props.label ? <h4 className="text-sm text-light-gray-2">{props.label}</h4> : null}
+            <SliderStyled
+              value={parseInt(value) || 0}
+              onChange={
+                ((_, value: number) => {
+                  setProp((props: any) => {
+                    if (isArray) {
+                      props[propKey][index] = onChange ? onChange(value) : value
+                    } else {
+                      props[propKey] = onChange ? onChange(value) : value
+                    }
+                  }, 1000)
+                }) as any
+              }
+            />
+          </>
+        ) : type === 'radio' ? (
+          <>
+            {props.label ? <h4 className="text-sm text-light-gray-2">{props.label}</h4> : null}
+            <RadioGroup
+              value={value || 0}
+              onChange={(e) => {
+                const value = e.target.value
+                setProp((props: any) => {
+                  props[propKey] = onChange ? onChange(value) : value
+                })
+              }}
+            >
+              {props.children}
+            </RadioGroup>
+          </>
+        ) : type === 'select' ? (
+          <ToolbarDropdown
+            value={value || ''}
+            onChange={(value) => setProp((props: any) => (props[propKey] = onChange ? onChange(value) : value))}
+            {...props}
+          />
+        ) : null}
```

Comment:
if type is required then it must be set. Instead of string conditions you could use a type or enum as a specific value. 

A switch would clean this conditional chaining like so 

```ts

enum ToolbarItemTypeEnum {
  SELECT = 'select',
  RADIO = 'radio',
  SLIDER = 'slider',
  NUMBER = 'number',
  TEXT = 'text',
  COLOR = 'color',
  BG = 'bg',
}

interface ToolbarItemProps {
  type: ToolbarItemTypeEnum,
}

const ToolbarItem = ({type}) => {
  switch (type) {
    case ToolbarItemTypeEnum.RADIO:
      return <>Radio</>
    case ToolbarItemTypeEnum.TEXT:
    case ToolbarItemTypeEnum.NUMBER:
    case ToolbarItemTypeEnum.COLOR:
      return <>the collection</>
    default:
      // No need for default return because type WILL be one of the cases
  }
}
```